components/input/gpio/gpio_test.go

```diff
@@ -502,21 +495,17 @@ func TestGPIOInput(t *testing.T) {
 		})
 	})
 
-	// Test poll frequency
+	// Test methodology: Issue n updates and ensure that each update is processed
+	// in the time based on the 'PollHz' config. e.g. a little over 100ms for 10hz.
+	// Note: This is a time-sensitive test and is prone to flakiness.
 	t.Run("axis poll frequency", func(t *testing.T) {
 		s := setup(t)
 		defer teardown(t, s)
 
-		s.b.Analogs["analog1"].Set(1023)
-		testutils.WaitForAssertion(t, func(tb testing.TB) {
-			tb.Helper()
-			state, err := s.dev.Events(s.ctx, map[string]interface{}{})
-			test.That(tb, err, test.ShouldBeNil)
-			test.That(tb, state["AbsoluteX"].Value, test.ShouldAlmostEqual, 1, 0.005)
-			test.That(tb, atomic.LoadInt64(&s.axis1Callbacks), test.ShouldEqual, 1)
-		})
+		epsilon := 15 * time.Millisecond
```

Comment:
Mild preference for moving this definition down to just above its first use. If it's up here, it looks like it's going to be used somewhere in the next 20 lines, but it's not.